src/zjs_fs.c

```diff
@@ -0,0 +1,940 @@
+                                     const jerry_value_t this,
+                                     const jerry_value_t argv[],
+                                     const jerry_length_t argc) {
+    return zjs_close(function_obj, this, argv, argc, 1);
+}
+
+static jerry_value_t zjs_unlink(const jerry_value_t function_obj,
+                                const jerry_value_t this,
+                                const jerry_value_t argv[],
+                                const jerry_length_t argc,
+                                uint8_t async)
+{
+    int ret = 0;
+
+    if (!jerry_value_is_string(argv[0])) {
+        ERR_PRINT("first parameter was not path\n");
+        return ZJS_UNDEFINED;
+    }
+    if (async) {
+        if (!jerry_value_is_function(argv[1])) {
+            ERR_PRINT("second parameter was not function\n");
+            return ZJS_UNDEFINED;
+        }
+    }
+
+    jerry_size_t jlen = jerry_get_string_size(argv[0]);
+    char path[jlen + 1];
+
+    int wlen = jerry_string_to_char_buffer(argv[0], (jerry_char_t *)path, jlen);
+    if (jlen != wlen) {
+        ERR_PRINT("size mismatch\n");
+        return ZJS_UNDEFINED;
+    }
+    path[wlen] = '\0';
+
+    ret = fs_unlink(path);
+
+    if (async) {
+        zjs_callback_id id = zjs_add_callback_once(argv[1],
+                                                   this,
+                                                   NULL,
+                                                   NULL);
+
+        jerry_value_t error = jerry_create_number(ret);
+        zjs_signal_callback(id, &error, sizeof(jerry_value_t));
+    }
+
+    return ZJS_UNDEFINED;
+}
+
+static jerry_value_t zjs_unlink_sync(const jerry_value_t function_obj,
+                                    const jerry_value_t this,
+                                    const jerry_value_t argv[],
+                                    const jerry_length_t argc) {
+    return zjs_unlink(function_obj, this, argv, argc, 0);
+}
+
+static jerry_value_t zjs_unlink_async(const jerry_value_t function_obj,
+                                     const jerry_value_t this,
+                                     const jerry_value_t argv[],
+                                     const jerry_length_t argc) {
+    return zjs_unlink(function_obj, this, argv, argc, 1);
+}
+
+static jerry_value_t zjs_read(const jerry_value_t function_obj,
+                             const jerry_value_t this,
+                             const jerry_value_t argv[],
+                             const jerry_length_t argc,
+                             uint8_t async)
+{
+    file_handle* handle;
+    int err = 0;
+
+    if (argc < 6 && async) {
+        ERR_PRINT("not enough parameters\n");
+        return ZJS_UNDEFINED;
+    }
+
+    if (!jerry_value_is_object(argv[0])) {
+        ERR_PRINT("first parameter was not file descriptor\n");
+        return ZJS_UNDEFINED;
+    }
+    if (!jerry_value_is_object(argv[1])) {
```

Comment:
Perhaps you should use zjs_buffer_find to make sure it's a buffer specifically? Not sure if we've been consistent about doing that. Maybe we should have a zjs_is_buffer().